namui-prebuilt/src/rect_slice/mod.rs

```diff
@@ -0,0 +1,270 @@
+            ($a: ident, $b: ident, $props_a: ident, $props_b: ident, $a_wh: ident, $b_xywh: ident) => {
+                namui::render![
+                    $a.render($a_wh, $props_a),
+                    namui::translate($b_xywh.x, $b_xywh.y, $b.render($b_xywh.wh(), $props_b))
+                ]
+            };
+        }
+
+        match self {
+            Slice::Left(a, b) => render_ab!(a, b, props_a, props_b, a_wh, b_xywh),
+            Slice::Right(a, b) => render_ab!(a, b, props_a, props_b, a_wh, b_xywh),
+            Slice::Top(a, b) => render_ab!(a, b, props_a, props_b, a_wh, b_xywh),
+            Slice::Bottom(a, b) => render_ab!(a, b, props_a, props_b, a_wh, b_xywh),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use self::traits::RectSlice;
+    use super::*;
+    use std::sync::atomic::AtomicBool;
+    use wasm_bindgen_test::wasm_bindgen_test;
+
+    #[test]
+    #[wasm_bindgen_test]
+    fn layout_should_give_right_wh() {
+        let button_render_called = AtomicBool::new(false);
+        let label_render_called = AtomicBool::new(false);
+        let body_render_called = AtomicBool::new(false);
+
+        let button = Column {
+            width: Size::from_parent_wh(&|parent_wh| parent_wh.height),
+            render: &|wh, _: ()| {
+                button_render_called.store(true, std::sync::atomic::Ordering::Relaxed);
+                assert_eq!(20.0, wh.width);
+                assert_eq!(20.0, wh.height);
+                RenderingTree::Empty
+            },
+        };
+        let label = Fill {
+            render: &|wh, _: ()| {
+                label_render_called.store(true, std::sync::atomic::Ordering::Relaxed);
+                assert_eq!(280.0, wh.width);
+                assert_eq!(20.0, wh.height);
+                RenderingTree::Empty
+            },
+        };
+
+        let header = Row {
+            height: Size::Fixed(20.0),
+            render: &|wh, props| Slice::Left(&button, &label).render(wh, props),
+        };
+
+        let body = Fill {
+            render: &|wh, _: ()| {
+                body_render_called.store(true, std::sync::atomic::Ordering::Relaxed);
+                assert_eq!(300.0, wh.width);
+                assert_eq!(480.0, wh.height);
+                RenderingTree::Empty
+            },
+        };
+
+        Slice::Top(&header, &body).render(
+            Wh {
+                width: 300.0,
+                height: 500.0,
+            },
+            (((), ()), ()),
+        );
+
+        assert_eq!(
+            true,
+            button_render_called.load(std::sync::atomic::Ordering::Relaxed)
+        );
+        assert_eq!(
+            true,
+            label_render_called.load(std::sync::atomic::Ordering::Relaxed)
+        );
+        assert_eq!(
+            true,
+            body_render_called.load(std::sync::atomic::Ordering::Relaxed)
+        );
+    }
+
+    #[test]
+    #[wasm_bindgen_test]
+    fn layout_with_traits_should_give_right_wh() {
```

Comment:
I like trait-way.

Reply:
You can use trait-way but if you want to use simple things without struct, then struct-way will be very helpful.